test/mocha/cli.js

```diff
@@ -94,6 +94,35 @@ describe("bin/uglifyjs", function () {
             done();
         });
     });
+    it("Should not load source map before finish reading from STDIN", function(done) {
+        var mapFile = "tmp/input.js.map";
+        try {
+            fs.mkdirSync("./tmp");
+        } catch (e) {
+            if (e.code != "EEXIST") throw e;
+        }
+        try {
+            fs.unlinkSync(mapFile);
+        } catch (e) {
+            if (e.code != "ENOENT") throw e;
+        }
+        var command = [
+            uglifyjscmd,
+            "--source-map", "content=" + mapFile,
+            "--source-map", "url=inline"
+        ].join(" ");
+
+        var child = exec(command, function(err, stdout) {
+            if (err) throw err;
+
+            assert.strictEqual(stdout, read("test/input/pr-3040/expect.js"));
+            done();
+        });
+        setTimeout(function() {
+            fs.copyFileSync("test/input/pr-3040/input.js.map", mapFile);
```

Comment:
My bad - didn't realise `fs.copyFile*` is only recently introduced in Node.js 8.5.0

Please change it to the following so it works on all supported versions of Node.js:
```js
fs.writeFileSync(mapFile, read("test/input/pr-3040/input.js.map"));
```

Reply:
Okay